cmd/cstor-pool-mgmt/volumereplica/volumereplica.go

```diff
@@ -699,24 +700,101 @@ func GetAndUpdateSnapshotInfo(
 	}
 
 	// If CVR is in rebuilding go and get snapshot information
-	// from other replicas
-	//if cvr.Status.Phase == apis.CVRStatusRebuilding {
-	//	err = getAndUpdatePendingSnapshotsList(clientset, cvr)
-	//}
+	// from other replicas and add snapshots under pending snapshot list
+	if cvr.Status.Phase == apis.CVRStatusRebuilding ||
+		cvr.Status.Phase == apis.CVRStatusReconstructingNewReplica {
+		err = getAndAddPendingSnapshotList(clientset, cvr)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update pending snapshots")
+		}
+	}
+
+	// It looks like hack but we must do this because of below reason
+	// - There might be chances in nth reconciliation CVR might be in Rebuilding
+	//   and pending snapshots added under CVR.Status.PendingSnapshots and after that
+	//   let us assume this pool is down meanwhile if snapshot deletion request
+	//   came and deleted snapshots in peer replicas. In next reconciliation if
+	//   CVR is Healthy then there might be chances that pending snapshots remains
+	//   as is to cover this corner case below check is required.
+	if cvr.Status.Phase == apis.CVRStatusOnline &&
+		len(cvr.Status.PendingSnapshots) != 0 {
+		cvr.Status.PendingSnapshots = nil
+	}
+	return nil
+}
+
+// getAndAddPendingSnapshotList get the snapshot information from peer replicas and
+// add under pending snapshot list
+// NOTE: Below function will delete the snapshot under pending snapshots if doesn't exists
+// on peer replicas
+func getAndAddPendingSnapshotList(
```

Comment:
add the logs for the snapnames that got added or deleted

Reply:
Done